react-app/components/FileManager/utils.js

```diff
@@ -0,0 +1,129 @@
+
+  const fileMapEntry = {
+    id: node.id,
+    name: node.name,
+    isDir: isDirectory,
+    metadata: node.metadata || {},
+    isSaved
+  };
+
+  if (isDirectory) {
+    fileMapEntry.childrenIds = [];
+    fileMapEntry.children = [];
+    if (node.children) {
+      for (const childNode of node.children) {
+        const childEntry = convertTreeToNewFileMap(childNode, fileMapEntry, newFileMapObject, isSaved);
+        fileMapEntry.childrenIds.push(childEntry.id);
+        fileMapEntry.children.push(childEntry);
+      }
+    }
+  }
+
+  if (parent) {
+    fileMapEntry.parentId = parent.id;
+  }
+
+  // Add the missing properties
+  if (fileMapEntry.isDir) {
+    fileMapEntry.childrenCount = fileMapEntry.childrenIds.length;
+  } else {
+    // Assuming you want to add size and modDate for files
+    if (node.metadata) {
+      fileMapEntry.size = node.metadata.file_size;
+      //fileMapEntry.modDate = node.metadata.modDate;
+    }
+  }
+
+  // Store the fileMapEntry in the newFileMapObject
+  newFileMapObject[node.id] = fileMapEntry;
+
+  return fileMapEntry;
+};
+
+/**
+ * Converts a tree structure to a new file map format.
+ *
+ * @param {Object} tree - The tree structure to convert.
+ * @param {boolean} isSaved - Indicates if the nodes are saved.
+ * @returns {Object} - The new file map format.
+ */
+export const convertTreeToNewFileMapFormat = (tree, isSaved = false) => {
+  const newFileMapObject = {};
+  const treeCloned = _.cloneDeep(tree);
+
+  convertTreeToNewFileMap(treeCloned, null, newFileMapObject, isSaved);
+
+  return newFileMapObject;
+};
+
+/**
+ * Converts a file map to a tree structure.
+ *
+ * @param {string} rootId - The ID of the root node.
+ * @param {string} parentPath - The path of the parent node.
+ * @param {Object} originalFileMapObject - The original file map object.
+ * @returns {Object} - The tree structure.
+ */
+export const convertFileMapToTree = (rootId, parentPath = '', originalFileMapObject) => {
+  const node = _.cloneDeep(originalFileMapObject[rootId]);
+
+  // Check if the node is a directory (has childrenIds) or a file (no childrenIds)
+  const isDirectory = node.isDir === true;
+  const nodePath = parentPath ? `${parentPath}/${node.name}` : node.name;
+
+  const tree = {
+    id: node.id,
+    name: node.name,
+    type: isDirectory ? 'directory' : 'file',
+    path: nodePath,
+    metadata: node.metadata || {},
+    parentId: node.parentId || ''
+  };
+
+  if (isDirectory && node.childrenIds) {
+    tree.children = [];
+    node.childrenIds.forEach((childId) => {
+      const childNode = convertFileMapToTree(childId, nodePath, originalFileMapObject);
+      tree.children.push(childNode);
+    });
+  }
+
+  return tree;
+};
+
+/**
+ * Finds a node by its ID in a tree structure.
+ *
+ * @param {Object} object - The tree structure to search.
+ * @param {string} id - The ID of the node to find.
+ * @returns {Object|null} - The found node or null if not found.
+ */
```

Comment:
[curious] Why are wen't using this type of docs elsewhere?

Reply:
It's not necessary to add jsdoc to all code